code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,412 @@
+			O.find_target()
+			O.update_explanation_text()
+
+			var/list/owners = O.get_owners()
+
+			if(!O.target && O.owner)
+				qdel(O)
+
+			for(var/M in owners)
+				var/datum/mind/own = M
+				to_chat(own.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+				O.owner.announce_objectives()
+
+// This function can not be undone; do not call this unless you are sure
+/obj/machinery/cryopod/proc/despawn_occupant()
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant.mind && mob_occupant.mind.assigned_role)
+		//Handle job slot/tater cleanup.
+		var/job = mob_occupant.mind.assigned_role
+		SSjob.FreeRole(job)
+		if(LAZYLEN(mob_occupant.mind.objectives))
+			mob_occupant.mind.objectives.Cut()
+			mob_occupant.mind.special_role = null
+
+	// Delete them from datacore.
+
+	var/announce_rank = null
+	for(var/medrecord in GLOB.data_core.medical)
+		var/datum/data/record/R = medrecord
+		if((R.fields["name"] == mob_occupant.real_name))
+			qdel(R)
+	for(var/secrecord in GLOB.data_core.security)
+		var/datum/data/record/T = secrecord
+		if((T.fields["name"] == mob_occupant.real_name))
+			qdel(T)
+	for(var/genrecord in GLOB.data_core.general)
+		var/datum/data/record/G = genrecord
+		if((G.fields["name"] == mob_occupant.real_name))
+			announce_rank = G.fields["rank"]
+			qdel(G)
+
+	//Make an announcement and log the person entering storage.
+	if(control_computer)
+		control_computer.frozen_crew += mob_occupant.real_name
+
+	if(GLOB.announcement_systems.len)
+		var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+		announcer.announce("CRYOSTORAGE", mob_occupant.real_name, announce_rank, list())
+		visible_message("<span class='notice'>\The [src] hums and hisses as it moves [mob_occupant.real_name] into storage.</span>")
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		if(W.loc.loc && (( W.loc.loc == loc ) || (W.loc.loc == control_computer)))
+			continue//means we already moved whatever this thing was in
+			//I'm a professional, okay
+		for(var/T in GLOB.typecache_cryoitems)
+			if(istype(W, T))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += W
+					mob_occupant.transferItemToLoc(W, control_computer, TRUE)
+				else
+					mob_occupant.transferItemToLoc(W, loc, TRUE)
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		qdel(W)	//because we moved all items to preserve away
+				//and yes, this totally deletes their bodyparts one by one, I just couldn't bother
+
+	if(iscyborg(mob_occupant))
+		var/mob/living/silicon/robot/R = occupant
+		if(!istype(R))
+			return
+		R.contents -= R.mmi
+		qdel(R.mmi)
+
+	mob_occupant.ghostize(FALSE)
+	handle_objectives()
+	QDEL_NULL(occupant)
+	for(var/obj/item/I in get_turf(src))
+		if(I in GLOB.typecache_cryoitems)
+			continue //Double safety check
+		qdel(I) //Cleanup anything left
```

Comment:
use is_type_in_typecache()

but.... why are we deleting all items on the turf?

Reply:
Suit storage is a bitch.